### DISCUSSION TITLE
Huge amounts of debug output emitted

### DISCUSSION THREAD

**Discussion by starpit** [micromark/micromark#102] [Q&A] [answered]
Hiya. We are attempting to upgrade from react-markdown v5 to v7. This upgrade seems to have pulled in a version of `micromark` that emits a huge amount of debug output. I mean **hundreds of thousands of lines of debug**. I usually run with `localStorage.debug='*'` when developing, which means that micromark slows things down hugely. Here are some examples

```
micromark main: passing `96` to start +0ms
micromark position: restore: `{"line":15,"column":1,"offset":1013,"_index":21,"_bufferIndex":0}` +0ms
micromark main: passing `96` to start +1ms
micromark enter: `codeFenced` +0ms
micromark enter: `codeFencedFence` +0ms
micromark enter: `codeFencedFenceSequence` +0ms
micromark consume: `96` +1ms
micromark main: passing `96` to sequenceOpen +0ms
micromark consume: `96` +0ms
```

This is mostly an FYI, i guess, because one can always use

```
localStorage.debug="*,-micromark"
```

but it seems odd to have such a huge amount of debug on without some extra flags?

**Comment by ChristianMurphy** [comment, marked as answer]
Thanks for reaching out, sorry you ran into a spot of trouble.

The verbose logs are hidden behind two flags. https://github.com/micromark/micromark#size--debug
1. one is an export condition which only is used in a development environment
2.  the other a `debug` package logging flag

In your example, both the export condition and the log level have been set to be verbose. 

If you don't want verbose debugging in development `localStorage.debug="*,-micromark"` could be a good option.
It will also not log when the export condition/environment changes to production.

**Reply by starpit** [reply]
Thanks for the kind reply. Yes, the `-micromark` is an adequate workaround. The downside is that we now need to add an important warning to our developer docs. I wish the `debug` npm had some support for debug levels. This extreme level of debug output would normally be relegated to a "SILLY" debug level. But we work with what we got!

**Reply by songkeys** [reply]
setting `localStorage.debug="*,-micromark"` does not work for me. i still got a flood of debug output:

<img width="449" height="176" alt="SCR-20251110-ftlu" src="https://github.com/user-attachments/assets/0f51f111-6c02-4d65-a45d-cb34c9286072" />
<img width="273" height="150" alt="SCR-20251110-fttr" src="https://github.com/user-attachments/assets/87311740-307a-424c-972c-e2f03d6c6fa5" />

Every time it outputs debug log, my chrome would freeze for a while... 

Do you have any suggestion on what to do? Can i completely disable it? I'm using vite to dev the server.
